packages/grid/_modules_/grid/hooks/features/rows/useRows.ts

```diff
@@ -87,10 +96,12 @@ export const useRows = (rows: RowsProp, apiRef: ApiRef): void => {
       }
 
       const idRowsLookup = allNewRows.reduce((lookup, row) => {
+        row = addRowId(getRowId, row);
+        checkRowHasId(row);
         lookup[row.id] = row;
         return lookup;
       }, {});
-      const allRows = allNewRows.map((row) => row.id);
+      const allRows = allNewRows.map((row) => getRowId(row));
```

Comment:
Could this boost performance?

```suggestion
      const allRows = getRowId ? allNewRows.map((row) => getRowId(row)) : allNewRows.map((row) => row.id);
```

Reply:
I don't think it will boost perf as `getRowId` returns `row.id` as it's just about removing 1 function of the memory. in exchange of a check for null value. So it will be negligeable, we can remove the default value but then we will have to do checks everywhere.
That said, getRowId is only used when initial rows are set or updated.